test/jdk/java/net/Socket/ConnectSocksProxyTest.java

```diff
@@ -0,0 +1,215 @@
+            assertThrows(IOException.class, () -> socket.connect(REFUSING_SOCKET_ADDRESS));
+            assertTrue(socket.isClosed());
+        }
+    }
+
+    /**
+     * Verifies that a bound socket is closed when {@code connect()} fails.
+     */
+    @Test
+    void testBoundSocket() throws IOException {
+        try (Socket socket = createProxiedSocket()) {
+            socket.bind(new InetSocketAddress(0));
+            assertTrue(socket.isBound());
+            assertFalse(socket.isConnected());
+            assertThrows(IOException.class, () -> socket.connect(REFUSING_SOCKET_ADDRESS));
+            assertTrue(socket.isClosed());
+        }
+    }
+
+    /**
+     * Verifies that a connected socket is not closed when {@code connect()} fails.
+     */
+    @Test
+    void testConnectedSocket() throws Throwable {
+        try (Socket socket = createProxiedSocket();
+             ServerSocket serverSocket = createEphemeralServerSocket()) {
+            socket.connect(serverSocket.getLocalSocketAddress());
+            try (Socket _ = serverSocket.accept()) {
+                assertTrue(socket.isBound());
+                assertTrue(socket.isConnected());
+                SocketException exception = assertThrows(
+                        SocketException.class,
+                        () -> socket.connect(REFUSING_SOCKET_ADDRESS));
+                assertEquals("Already connected", exception.getMessage());
+                assertFalse(socket.isClosed());
+            }
+        }
+    }
+
+    /**
+     * Delegates to {@link #testUnconnectedSocketWithUnresolvedAddress(boolean, Socket)} using an unbound socket.
+     */
+    @Test
+    void testUnboundSocketWithUnresolvedAddress() throws IOException {
+        try (Socket socket = createProxiedSocket()) {
+            assertFalse(socket.isBound());
+            assertFalse(socket.isConnected());
+            testUnconnectedSocketWithUnresolvedAddress(false, socket);
+        }
+    }
+
+    /**
+     * Delegates to {@link #testUnconnectedSocketWithUnresolvedAddress(boolean, Socket)} using a bound socket.
+     */
+    @Test
+    void testBoundSocketWithUnresolvedAddress() throws IOException {
+        try (Socket socket = createProxiedSocket()) {
+            socket.bind(new InetSocketAddress(0));
+            testUnconnectedSocketWithUnresolvedAddress(true, socket);
+        }
+    }
+
+    /**
+     * Verifies the behaviour of an unconnected socket when {@code connect()} is invoked using an unresolved address.
+     */
+    private static void testUnconnectedSocketWithUnresolvedAddress(boolean bound, Socket socket) throws IOException {
+        assertEquals(bound, socket.isBound());
+        assertFalse(socket.isConnected());
+        try (ServerSocket serverSocket = createEphemeralServerSocket()) {
+            InetSocketAddress unresolvedAddress =
+                    InetSocketAddress.createUnresolved("localhost", serverSocket.getLocalPort());
```

Comment:
```suggestion
                    InetSocketAddress.createUnresolved(getLoopbackAddress().getHostAddress(), serverSocket.getLocalPort());
```

Reply:
Fixed in 9706ef9c7c362829daa7cc3ea845a85e36bf92c4.